src/filemap.rs

```diff
@@ -36,16 +36,95 @@ pub fn write_all_files(file_map: &FileMap,
                        config: &Config)
                        -> Result<(HashMap<String, String>), io::Error> {
     let mut result = HashMap::new();
+    output_heading(mode).ok();
+
     for filename in file_map.keys() {
         let one_result = try!(write_file(&file_map[filename], filename, mode, config));
         if let Some(r) = one_result {
             result.insert(filename.clone(), r);
         }
     }
+    output_trailing(mode).ok();
 
     Ok(result)
 }
 
+pub fn output_heading(mode: WriteMode) -> Result<(), io::Error> {
```

Comment:
Oh it's for all file, sorry brain fart, I thought it was for each file. With that said, for handling 1 special case I'd prefer if mode == WriteMode::Checkstyle { try!(output_trailing()); }. Especially when we don't need to do anything for all other cases.